.github/workflows/index-resources.yml

```diff
@@ -17,19 +17,76 @@ on:
         type: string
         required: true
 
+defaults:
+  run:
+    # This is the same as GitHub Action's `bash` keyword as of 20 June 2023:
+    # https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell
+    #
+    # Completely spelling it out here so that GitHub can't change it out from under us
+    # and we don't have to refer to the docs to know the expected behavior.
+    shell: bash --noprofile --norc -eo pipefail {0}
+
 jobs:
+  production_app:
+    if: ${{ inputs.ref == '' }}
+    uses: ./.github/workflows/get-heroku-app-commit-index.yml
+    secrets: inherit
+    with:
+      app_name: "nextstrain-server"
+
+  canary_app:
+    if: ${{ inputs.ref == '' }}
+    uses: ./.github/workflows/get-heroku-app-commit-index.yml
+    secrets: inherit
+    with:
+      app_name: "nextstrain-canary"
+
+  build_ref_matrix:
+    needs: [production_app, canary_app]
+    if: ${{ !cancelled() }}
+    runs-on: ubuntu-latest
+    outputs:
+      ref_matrix: ${{ steps.ref_matrix.outputs.ref_matrix }}
+    env:
+      INPUT_REF: ${{ inputs.ref }}
+      PRODUCTION_COMMIT: ${{ needs.production_app.outputs.commit_hash }}
+      PRODUCTION_RESOURCE_INDEX: ${{ needs.production_app.outputs.resource_index }}
+      CANARY_COMMIT: ${{ needs.canary_app.outputs.commit_hash }}
+      CANARY_RESOURCE_INDEX: ${{ needs.canary_app.outputs.resource_index }}
+    steps:
+      - id: ref_matrix
+        name: Create ref matrix
+        run: |
+          if [[ "$INPUT_REF" != '' ]]; then
+            jq_filter='[ env.INPUT_REF ]'
+          elif [[ "$PRODUCTION_RESOURCE_INDEX" == "$CANARY_RESOURCE_INDEX" ]]; then
+            jq_filter='[ env.PRODUCTION_COMMIT ]'
+          else
+            jq_filter='[ env.PRODUCTION_COMMIT, env.CANARY_COMMIT ]'
+          fi
+
+          ref_matrix=$(jq -c --null-input "$jq_filter")
```

Comment:
Ah, bad naming on my part! Like you said, this is just my way of ensuring I am building a properly formatted JSON array.

Reply:
In case it's clarifying, the difference between:

```bash
ref_matrix=$(jq -c --null-input "[ env.PRODUCTION_COMMIT, env.CANARY_COMMIT ]")
```

and something like:

```bash
ref_matrix="[ \"$PRODUCTION_COMMIT\", \"$CANARY_COMMIT\" ]"
```

is that the former ensures a valid JSON value and the latter does not.